internal/docstore/dynamodocstore/dynamo.go

```diff
@@ -12,22 +12,134 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Package dynamodocstore provides a docstore implementation backed by AWS
+// DynamoDB.
+// Use OpenCollection to construct a *docstore.Collection.
+//
+// URLs
+//
+// For docstore.OpenCollection, dynamodocstore registers for the scheme
+// "dynamodocstore". The default URL opener will use an AWS session with the
+// default credentials and configuration; see
+// https://docs.aws.amazon.com/sdk-for-go/api/aws/session/ for more details.
+// To customize the URL opener, or for more details on the URL format, see
+// URLOpener.
+// See https://godoc.org/gocloud.dev#hdr-URLs for background information.
 package dynamodocstore
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"net/url"
 	"strings"
+	"sync"
 
 	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/session"
 	dyn "github.com/aws/aws-sdk-go/service/dynamodb"
 	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
+	gcaws "gocloud.dev/aws"
 	"gocloud.dev/gcerrors"
 	"gocloud.dev/internal/docstore"
 	"gocloud.dev/internal/docstore/driver"
 	"gocloud.dev/internal/gcerr"
 )
 
+func init() {
+	docstore.DefaultURLMux().RegisterCollection(Scheme, new(lazySessionOpener))
+}
+
+type lazySessionOpener struct {
+	init   sync.Once
+	opener *URLOpener
+	err    error
+}
+
+func (o *lazySessionOpener) OpenCollectionURL(ctx context.Context, u *url.URL) (*docstore.Collection, error) {
+	o.init.Do(func() {
+		sess, err := session.NewSessionWithOptions(session.Options{SharedConfigState: session.SharedConfigEnable})
+		if err != nil {
+			o.err = err
+			return
+		}
+		o.opener = &URLOpener{
+			ConfigProvider: sess,
+		}
+	})
+	if o.err != nil {
+		return nil, fmt.Errorf("open collection %s: %v", u, o.err)
+	}
+	return o.opener.OpenCollectionURL(ctx, u)
+}
+
+// Scheme is the URL scheme dynamodocstore registers its URLOpener under on
+// docstore.DefaultMux.
+const Scheme = "dynamodocstore"
+
+// URLOpener opens dynamodocstore URLs like
+// "dynamodocstore://mytable?partition_key=partkey&sort_key=sortkey".
+//
+// The URL Host is used as the table name. See
```

Comment:
Same question here as for Mongo: Is "foo/bar" a valid table name?

Reply:
No, I updated the link below directly to the naming rules.